src/coreclr/vm/ceeload.cpp

```diff
@@ -345,6 +345,7 @@ Module::Module(Assembly *pAssembly, PEAssembly *pPEAssembly)
     m_pAssembly = pAssembly;
     m_pPEAssembly      = pPEAssembly;
     m_dwTransientFlags = CLASSES_FREED;
+    m_dwPersistedFlags = 0x00000001;
```

Comment:
Could you please explain how the fix works and what the 0x00000001 constant represents?

Reply:
Gladly.  In trying the steps in the following article:
https://learn.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting?source=docs
in the call to load_assembly_and_get_function_pointer, it calls MethodTable *Module::GetGlobalMethodTable()
m_dwPersistedFlags was uninitialized = ??? causing the code to simply hard crash.

I chose 0x00000001 because its the unused example as such.  I'm not sure if this is per standard but it never was initialized to a default value.

The enum is as follows:
    enum {
        // These are the values set in m_dwPersistedFlags.
        // unused                   = 0x00000001,
        COMPUTED_GLOBAL_CLASS       = 0x00000002,

        // unused                   = 0x00000004,
        // unused                   = 0x00000008,

        // This flag applies to assembly, but it is stored so it can be cached in ngen image
        COMPUTED_WRAP_EXCEPTIONS    = 0x00000010,
        WRAP_EXCEPTIONS             = 0x00000020,

        // unused                   = 0x00000040,

        // This flag applies to assembly, but is also stored here so that it can be cached in ngen image
        COLLECTIBLE_MODULE          = 0x00000080,

        //If attribute value has been cached before
        DEFAULT_DLL_IMPORT_SEARCH_PATHS_IS_CACHED   = 0x00000400,

        //If module has default dll import search paths attribute
        DEFAULT_DLL_IMPORT_SEARCH_PATHS_STATUS      = 0x00000800,


        //If setting has been cached
        RUNTIME_MARSHALLING_ENABLED_IS_CACHED = 0x00008000,
        //If runtime marshalling is enabled for this assembly
        RUNTIME_MARSHALLING_ENABLED = 0x00010000,
    };

Please let me know if that will suffice.  Thanks, Ken

Reply:
This memory is expected to be zero-filled as the comment on the next line says. It is expected to be pre-initialized to 0 by the time this code is executed. It should not be necessary to initialize like this.

Something else went wrong in your test.